tests/PHPStan/Rules/Classes/ImpossibleInstanceOfRuleTest.php

```diff
@@ -368,4 +368,24 @@ public function testBug5333(): void
 		]);
 	}
 
+	public function testBug3632(): void
+	{
+		$this->checkAlwaysTrueInstanceOf = true;
+		$this->treatPhpDocTypesAsCertain = false;
+		$this->analyse([__DIR__ . '/data/bug-3632.php'], []);
+	}
+
+	public function testTraitInstanceOf(): void
+	{
+		$this->checkAlwaysTrueInstanceOf = true;
+		$this->treatPhpDocTypesAsCertain = false;
+
+		$this->analyse([__DIR__ . '/../../Analyser/data/trait-instance-of.php'], [
+			[
+				'Instanceof between $this(TraitInstanceOf\ATrait1Class) and trait TraitInstanceOf\Trait2 will always evaluate to false.',
```

Comment:
for some reason this error is not reported. I get a error though, when running phpstan on this file in isolation.